crates/parser/src/options.rs

```diff
@@ -1,10 +1,72 @@
 use std::collections::BTreeMap;
 use std::fmt;
+use std::str::FromStr;
 
-use datafusion::common::parsers::CompressionTypeVariant;
-use datafusion::common::FileType;
-use datafusion::sql::sqlparser::parser::ParserError;
-use protogen::metastore::types::options::StorageOptions;
+use sqlparser::parser::ParserError;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum FileType {
+    /// Apache Arrow file
+    ARROW,
+    /// Apache Avro file
+    AVRO,
+    /// Apache Parquet file
+    PARQUET,
+    /// CSV file
+    CSV,
+    /// JSON file
+    JSON,
+}
```

Comment:
this is (I think?) doesn't impact the similarly named enum in data fusion (because of the dependency between `FileType` and `FileFormat`... 

having a noe on this somewhere might be good, otherwise things become difficult to grep and make sense of things.

Reply:
This is a direct copy from datafusion's `FileType` for now. I needed to copy it over to break our dependency in `parser`. Will add a note that it's a direct equivalent to datafusion's `FileType`.